core/dbt/adapters/base/impl.py

```diff
@@ -237,24 +281,42 @@ def _relations_filter_table(cls, table, schemas):
         """
         return table.where(_relations_filter_schemas(schemas))
 
+    def _get_cache_schemas(self, manifest):
+        """Get a mapping of each node's "information_schema" relations to a
+        set of all schemas expected in that information_schema.
+
+        There may be keys that are technically duplicates on the database side,
+        for example all of '"foo", 'foo', '"FOO"' and 'FOO' could coexist as
+        databases, and values could overlap as appropriate. All values are
+        lowercase strings.
+        """
+        info_schema_name_map = SchemaSearchMap()
+        for node in manifest.nodes.values():
+            if node.resource_type in NodeType.executable():
```

Comment:
i think we want to include _all_ nodes (or at least, add Sources) here. As a result of this code, i think the `get_catalog` macro doesn't correctly loop over databases for sources! 

I made a change over here (https://github.com/fishtown-analytics/dbt/pull/1332/files#diff-f6971115b03b21cf1f245c037cda4978R239) which only cached populated the cache for schemas that dbt was going to try to build models into. The _single_ place we need to loop over Source databases is in the `get_catalog` macro, as we use it to build Sources into the docs site. Otherwise, dbt shouldn't care about the databases provided in Sources